Destiny.ScrimTracker.Api/Controllers/GuardianController.cs

```diff
@@ -2,58 +2,97 @@
             _guardianService = guardianService;
             _logger = logger;
         }
 
         [HttpGet]
-        public IEnumerable<Guardian> Get()
+        public IActionResult Get()
         {
-            return _guardianService.GetGuardians();
+            var guardianSnapshots = _guardianService.GetGuardians();
+            return View(guardianSnapshots);
         }
 
-        [HttpGet("{guardianId}")]
-        public Guardian GetGuardian(string guardianId)
+        [HttpGet("/new")]
+        [Authorize]
+        public IActionResult AddGuardianView()
         {
-            return _guardianService.GetGuardian(guardianId);
+            return View();
         }
-
-        [HttpPost]
-        public string Post(CreateGuardianRequest request)
+        
+        [HttpPost("/new")]
+        [Authorize]
+        public IActionResult AddGuardian([FromForm] CreateGuardianRequest request)
         {
+            if (!ModelState.IsValid)
+            {
+                return RedirectToAction("AddGuardianView");
+            }
+            
             var guardian = request.ToGuardian();
-            return _guardianService.CreateGuardian(guardian);
+            _guardianService.CreateGuardian(guardian);
+            return Redirect($"/guardians");
+        }
+        
+        [HttpGet("{guardianId}")]
+        public IActionResult GetGuardian(string guardianId)
+        {
+            var guardian = _guardianService.GetGuardian(guardianId);
+            var guardianElo = _guardianService.GetGuardianElo(guardianId);
+            var guardianEfficiency = _guardianService.GetGuardianEfficiency(guardianId);
+            var matchCount = _matchService.GetMatchResultsForGuardian(guardianId).Count();
+            
+            var guardianHistory = new GuardianHistory()
+            {
+                Guardian = guardian,
+                EfficiencyHistory = guardianEfficiency,
+                EloHistory = guardianElo,
+                MatchCount = matchCount
+            };
+            
+            return View(guardianHistory);
         }
 
         [HttpPut("{guardianId}")]
-        public Guardian Put(UpdateGuardianRequest request, [FromRoute] string guardianId)
+        [Authorize]
+        public IActionResult Put(UpdateGuardianRequest request, [FromRoute] string guardianId)
         {
             var guardian = request.ToGuardian(guardianId);
-            return _guardianService.UpdateGuardian(guardian);
+            var updatedGuardian = _guardianService.UpdateGuardian(guardian);
+
+            return Redirect($"/guardians/{updatedGuardian.Id}");
         }
 
         [HttpDelete("{guardianId}")]
-        public string Delete([FromRoute] string guardianId)
+        [Authorize]
+        public IActionResult Delete([FromRoute] string guardianId)
         {
-            return _guardianService.DeleteGuardian(guardianId);
+            var guardian = _guardianService.DeleteGuardian(guardianId);
+            return Json(guardian);

```

Comment:
Why JSON? I believe a "normal" `DELETE` redirects to the list of resources, eg `guardians`

Reply:
Aye. Couldn't figure out how to change the redirect HTTP method. was trying to send the request as delete, so i had this return, and the AJAX call handles the redirect